web/src/settings_invites.ts

```diff
@@ -35,27 +56,23 @@ function failed_listing_invites(xhr) {
     );
 }
 
-function add_invited_as_text(invites) {
+function add_invited_as_text(invites: Invite[]): void {
     for (const data of invites) {
         data.invited_as_text = settings_config.user_role_map.get(data.invited_as);
     }
 }
 
-function sort_invitee(a, b) {
+function sort_invitee(a: Invite, b: Invite): number {
     // multi-invite links don't have an email field,
     // so we set them to empty strings to let them
     // sort to the top
-    const str1 = (a.email || "").toUpperCase();
-    const str2 = (b.email || "").toUpperCase();
+    const str1 = a.is_multiuse ? "" : a.email.toUpperCase();
+    const str2 = b.is_multiuse ? "" : b.email.toUpperCase();
 
     return util.strcmp(str1, str2);
 }
 
-function populate_invites(invites_data) {
-    if (!meta.loaded) {
-        return;
-    }
```

Comment:
Or is there supposed to be some event wired up from `server_events_dispatch` to this (currently non-exported) function, like there is for `settings_exports.populate_exports`, `settings_profile_fields.populate_profile_fields`, etc.?

Reply:
I think this very inefficient thing in `server_events_dispatch` is how that gets called:

```
        case "invites_changed":                                                                         
            if ($("#admin-invites-list").length) {                                                      
                settings_invites.set_up(false);                                                         
            }                                                                                           
            break;                                                                                      
```

I don't think we actually care that it's inefficient but that is a mechanism through which it gets called more than once.